src/common/Microsoft.DotNet.UpgradeAssistant.Abstractions/AnalyzeResult.cs

```diff
@@ -7,8 +7,8 @@ namespace Microsoft.DotNet.UpgradeAssistant.Analysis
 {
     public record AnalyzeResult
     {
-        public string AnalysisFileLocation { get; init; }
+        public string? AnalysisFileLocation { get; init; }
 
-        public IReadOnlyCollection<string> AnalysisResults { get; init; }
+        public IReadOnlyCollection<string>? AnalysisResults { get; init; }
```

Comment:
Instead of having it nullable, set it to `Array.Empty<string>` first so you don't have to handle any nulls